### PR TITLE
build(pnpm): basic setup for pnpm

### PR DESCRIPTION
**Pull request by GermainBergeron**
### Proposed Changes

Delete package-lock.json files
Setup pnpm workspace and lock file
Bump dependencies version to match across multiple packages
Add missing dependencies

### Potential Breaking Changes

None for now

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/react-vapor/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

File: packages/demo/webpack.config.js [hunk 1]
```diff
@@ -44,7 +44,10 @@ module.exports = {
         new webpack.ContextReplacementPlugin(/moment[/\\]locale$/, /en-ca/),
         new ForkTsCheckerWebpackPlugin({
             typescript: {
-                build: true,
+                // build: true,
```
Comment (gdostie) [inline packages/demo/webpack.config.js:47]: Remove comment?

Comment (gdostie) [reply packages/demo/webpack.config.js:47]: Also did you try installing the `@types/node` package instead of removing the `build: true` option here?

**Review by gdostie — APPROVED** [review]

File: packages/demo/webpack.config.js [hunk 2]
```diff
@@ -30,7 +30,7 @@ module.exports = {
         new CopyPlugin({
             patterns: [
                 path.resolve(__dirname, 'node_modules', 'jquery', 'dist', 'jquery.slim.min.js'),
-                path.resolve(__dirname, '..', '..', 'node_modules', 'chosen-js', 'chosen.jquery.min.js'),
+                path.resolve(__dirname, 'node_modules', 'react-vapor', 'node_modules', 'chosen-js', 'chosen.jquery.min.js'),
```
Comment (rosalie-liu) [inline packages/demo/webpack.config.js:33]: We have 2 `node_modules` in this line, is it wanted? :thinking:

**Review by rosalie-liu — APPROVED** [review]

**Review by aiheon — APPROVED** [review]

File: packages/demo/webpack.config.js (hunk 2 above)
Comment (GermainBergeron) [reply packages/demo/webpack.config.js:33]: Yes, [pnpm doesn't use a flat node_modules structure](https://pnpm.io/5.x/symlinked-node-modules-structure)

File: packages/demo/webpack.config.js (hunk 1 above)
Comment (GermainBergeron) [reply packages/demo/webpack.config.js:47]: Removed.
Yes I tried, but it does not work (I get the same errors)

**Pull request merged**